Add ADOPTERS.md file

estesp opened the pull request:
Per request of the CNCF, adding known containerd adopters to GitHub
project.

Signed-off-by: Phil Estes <estesp@linux.vnet.ibm.com>

estesp commented:
//cc: @caniszczyk

caniszczyk commented:
+1

You may want to add a blurb in the README file linking to this and encourage folks to add themselves via a PR.

jterry75 commented:
@estesp - I don't think it matters but should we differentiate between CRI and containerd API in this case? I know its 99% the same code but wondering if the CNCF wanted to know about the entry point API or not?

LGTM

estesp commented:
@jterry75 I guess my thinking is that this list is about consumption of containerd in general, which is of interest to the CNCF for their projects. Given the CRI plugin is part of the overall containerd project, I don't think it matters from an adopters list standpoint.

I am trying to mention Kubernetes and/or the CRI for adopters who are using it via the CRI plugin model.

AkihiroSuda commented on ADOPTERS.md:
```diff
@@ -0,0 +1,31 @@
+
+A non-exhaustive list of containerd adopters is provided below.
+
+**_Docker engine_** - Containerd began life prior to its CNCF adoption as a lower-layer
```
nit: s/Docker/Moby and Docker/ ?

estesp commented:
I have added a link from `README.md` to the adopters list (and encourage PRs to update it), and updated the Docker line to include a comment about Moby (`moby/moby`) engine project uses @AkihiroSuda

AkihiroSuda commented on ADOPTERS.md:
```diff
@@ -0,0 +1,33 @@
+
+**_Azure acs-engine_** - Microsoft Azure's [acs-engine](https://github.com/Azure/acs-engine) open source project has customizable deployment of Kubernetes clusters, where containerd is a selectable container runtime. At some point in the future Azure's AKS service will default to use containerd as the CRI runtime for deployed Kubernetes clusters.
+
+**_Other Projects_** - While the above list provides a cross-section of well known uses of containerd, the simplicity and clear API layer for containerd has inspired many smaller projects around providing simple container management platforms. Two that have come from containerd community participants directly are Michael Crosby's [boss](https://github.com/crosbymichael/boss) project and Evan Hazl ...
```
https://github.com/godarch/darch cc @pauldotknopf

estesp replied:
Thanks; good reminder! Forgot about that project--I've added it now

mikebrow commented on ADOPTERS.md:
```diff
@@ -0,0 +1,33 @@
+
+A non-exhaustive list of containerd adopters is provided below.
+
+**_Docker/Moby engine_** - Containerd began life prior to its CNCF adoption as a lower-layer
```
link to docker?

mikebrow commented on ADOPTERS.md:
```diff
@@ -0,0 +1,33 @@
+includes any use of the open source Moby engine project; including the Balena project
+listed below.
+
+**_IBM Cloud Kubernetes Service (IKS)_** - offers containerd as the CRI runtime for v1.11 and, soon, v1.12.
```
link..

mikebrow replied:
suggest removing version data beyond just saying vx.x+ as that will need require frequent updates..

mikebrow commented on ADOPTERS.md:
```diff
@@ -0,0 +1,33 @@
+
+**_IBM Cloud Kubernetes Service (IKS)_** - offers containerd as the CRI runtime for v1.11 and, soon, v1.12.
+
+**_IBM Cloud Private (ICP)_** - IBM's on-premises cloud offering has containerd as a "tech preview" CRI runtime for the Kubernetes offered within this product for the past two releases, and plans to fully migrate to containerd in a future release.
```
link.. agree leave preview data till fully supported.

mikebrow commented on ADOPTERS.md:
```diff
@@ -0,0 +1,33 @@
+
+**_IBM Cloud Private (ICP)_** - IBM's on-premises cloud offering has containerd as a "tech preview" CRI runtime for the Kubernetes offered within this product for the past two releases, and plans to fully migrate to containerd in a future release.
+
+**_Google Cloud Kubernetes Engine (GKE)_** - offers containerd as the CRI runtime in **beta** for recent versions of Kubernetes.
```
link..

mikebrow commented on ADOPTERS.md:
```diff
@@ -0,0 +1,33 @@
+
+**_Azure acs-engine_** - Microsoft Azure's [acs-engine](https://github.com/Azure/acs-engine) open source project has customizable deployment of Kubernetes clusters, where containerd is a selectable container runtime. At some point in the future Azure's AKS service will default to use containerd as the CRI runtime for deployed Kubernetes clusters.
+
+**_Other Projects_** - While the above list provides a cross-section of well known uses of containerd, the simplicity and clear API layer for containerd has inspired many smaller projects around providing simple container management platforms. Two that have come from containerd community participants directly are Michael Crosby's [boss](https://github.com/crosbymichael/boss) project and Evan Hazl ...
```
suggest list format for these

cpuguy83 commented on ADOPTERS.md:
```diff
@@ -0,0 +1,33 @@
+
+**_BuildKit_** - The Moby project's [BuildKit](https://github.com/moby/buildkit) can use either runC or containerd as build execution backends for building container images. BuildKit support has also been built into the Docker engine in recent releases, making BuildKit provide the backend to the `docker build` command.
+
+**_Azure acs-engine_** - Microsoft Azure's [acs-engine](https://github.com/Azure/acs-engine) open source project has customizable deployment of Kubernetes clusters, where containerd is a selectable container runtime. At some point in the future Azure's AKS service will default to use containerd as the CRI runtime for deployed Kubernetes clusters.
```
Maybe best to leave off the last bit?

cpuguy83 replied:
Note: Don't take this as some official request, just that it hasn't happened so it's just speculation here.

estesp replied:
It is a statement from someone in Azure leadership who said I could say it publicly. Wasn't sure whether to quote them or just leave it with the more vague: "at some point in the future"

crosbymichael commented:
LGTM

estesp commented:
Thanks for the review @mikebrow; I believe I responded to each in the latest update.
